### PR TITLE
Detect maintenance mode #4485

### PR DESCRIPTION
**Pull request by ckamm**
When we first detect a 503 (probably from a PROPFIND) and enter the
ServiceUnavailable state, we new trigger a status.php query that will
switch the state to MaintenanceMode if necessary.

### CODE REVIEW

File: src/gui/accountstate.cpp [hunk 1]
```diff
@@ -94,6 +94,14 @@ void AccountState::setState(State state)
             _connectionStatus = ConnectionValidator::Undefined;
             _connectionErrors.clear();
         } else if (oldState == SignedOut && _state == Disconnected) {
+            // If we stop being voluntarily signed-out, try to connect and
```
Comment (guruz) [inline src/gui/accountstate.cpp]: Is this related to the maintenance mode?

Comment (ckamm) [inline src/gui/accountstate.cpp]: Nah, I just sneaked in this comment because I felt it was missing. It's technically unrelated.

**Comment by SamuAlfageme** [comment]
@ckamm I'm seeing `AccountState::ServiceUnavailable`'s Connection Label (`Server <server> is temporarily unavailable.`) on the account tab after setting the maintenance mode in oC 10.0.2. The value is correctly set in `status.php` but the client seems to be ignoring it 😕  

(I'm on [`2.4.0git 03aa7e`](https://github.com/owncloud/client/commit/03aa7e30449e6777fc23a96ff315020ed33da05b) revision)

**Comment by ckamm** [comment]
@SamuAlfageme You're *not* seeing it, I guess? Will test again.

**Comment by SamuAlfageme** [comment]
@ckamm oh, no. I mean by looking at 0ba33e0f399db04f6e5cc4a5fed99058e719a36f:

```diff
+ } else if (state == AccountState::MaintenanceMode) {
+            showConnectionLabel( tr("Server %1 is currently in maintenance mode.").arg(server) );
```

I thought `Server <server> is currently in maintenance mode.` is the label to expect there.

**Comment by ckamm** [comment]
@SamuAlfageme Yep, I misread your comment. I can reproduce it and am surprised about that.

**Comment by ckamm** [comment]
@SamuAlfageme Fixed, got broken when we switched to Qt json parser.
